### PR TITLE
Enhance support for wasm testing with emscripten test suite

### PR DESCRIPTION
**Pull request by dschuff**
1) Add support for running tests with d8's native wasm support (instead
of the binaryen interpreter)
2) Skip setjmp/longjmp and exception handling tests when using the wasm
backend, as these are not supported yet.

### CODE REVIEW

File: tests/runner.py [hunk 1]
```diff
@@ -602,6 +602,8 @@ def do_run_from_file(self, src, expected_output, args=[], output_nicerizer=None,
 
   ## Does a complete test - builds, runs, checks output, etc.
   def do_run(self, src, expected_output, args=[], output_nicerizer=None, output_processor=None, no_build=False, main_file=None, additional_files=[], js_engines=None, post_build=None, basename='src.cpp', libraries=[], includes=[], force_c=False, build_ll_hook=None, extra_emscripten_args=[], assert_returncode=None):
+    if Settings.DISABLE_EXCEPTION_CATCHING == 0 and self.is_wasm_backend():
+      return self.skip("Wasm backend doesn't support exceptions yet")
```
Comment (kripken) [inline tests/runner.py]: either WebAssembly or wasm (not capitalized Wasm)

File: tools/jsrun.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def make_command(filename, engine=None, args=[]):
   # label a path to nodejs containing a 'd8' as spidermonkey instead.
   jsengine = os.path.split(engine[0])[-1]
   # Use "'d8' in" because the name can vary, e.g. d8_g, d8, etc.
-  return engine + [filename] + (['--'] if 'd8' in jsengine or 'jsc' in jsengine else []) + args
+  return engine + [filename] + (['--expose-wasm', '--'] if 'd8' in jsengine or 'jsc' in jsengine else []) + args
```
Comment (kripken) [inline tools/jsrun.py]: I guess this would fail if v8 isn't new enough? that's probably ok though, by default we test in only node.

**Comment by kripken** [comment]
lgtm with minor comments

File: tools/jsrun.py (hunk 2 above)
Comment (dschuff) [inline tools/jsrun.py]: Yeah, and the flag has been in d8 for several months, so I don't think it should be a big issue

File: tests/runner.py (hunk 1 above)
Comment (dschuff) [inline tests/runner.py]: Done

**Pull request merged**